crates/sui-core/src/authority.rs

```diff
@@ -239,6 +239,12 @@ impl AuthorityState {
         &self,
         transaction: Transaction,
     ) -> Result<TransactionInfoResponse, SuiError> {
+        // Validators should never sign an external system transaction.
+        fp_ensure!(
+            !transaction.data.kind.is_system_tx(),
+            SuiError::InvalidSystemTransaction
+        );
```

Comment:
it might be good to have a test that ensures this check holds - losing this check during a refactoring could be disastrous